ui/appui-react/src/appui-react/selection/HideIsolateEmphasizeManager.ts

```diff
@@ -340,9 +352,10 @@ export class HideIsolateEmphasizeManager extends HideIsolateEmphasizeActionHandl
   private static async getSelectionSetElementCategories(
     iModel: IModelConnection
   ) {
-    const props = (await iModel.elements.getProps(
-      iModel.selectionSet.elements
-    )) as GeometricElementProps[];
+    const props =
+      (await HideIsolateEmphasizeManager.getSelectionSetElementProps(
+        iModel
+      )) as GeometricElementProps[];
```

Comment:
We are getting lucky with this typecast, since `prop.category` is false if the required property doesn't exist...

Reply:
What is the problem here? If it returns false, then the category from the prop does not get added to the set so it will be irrelevant but nothing will go wrong.

Reply:
Yes, if we are certain that only geometric elements are in the selection set when transient ids are filtered out.

Reply:
Decided to leave it as it is for now as there seems to be no simple way to ensure that the element is geometric. Possible alternative would be something like this: https://github.com/iTwin/DevicePainter/blob/ee0af3202c8faaea7dcaaef269c73b9e1783a16e/src/frontend/api/IModelHelper.ts#L140